src/main/kotlin/org/rust/ide/inspections/RsAssertEqualInspection.kt

```diff
@@ -0,0 +1,90 @@
+
+                            override fun getFamilyName(): String = name
+
+                            override fun applyFix(project: Project, descriptor: ProblemDescriptor) {
+                                //Extract Compared Assert Argument
+                                val macro = descriptor.psiElement as RsMacroCall
+                                val assertArg = macro.assertMacroArgument!!
+                                val argPair = comparedAssertArgs(assertArg) ?: return
+
+                                //Build assert_eq
+                                val factory = RsPsiFactory(project)
+                                val newAssert = factory.createExpression("assert_eq!()") as RsCallExpr
+                                val a = newAssert.valueArgumentList.addAfter(argPair.first, newAssert.valueArgumentList.lparen)
+                                if (argPair.second != null) {
+                                    val comma = newAssert.valueArgumentList.addAfter(factory.createComma(), a)
+                                    newAssert.valueArgumentList.addAfter(argPair.second!!, comma)
+                                }
+                                macro.replace(newAssert)
+                            }
+                        }
+                    )
+                }
+                ComparisonOp.EXCLEQ -> {
+                    holder.registerProblem(
+                        o,
+                        "assert!(a != b) can be assert_ne!(a, b)",
+                        object : LocalQuickFix {
+                            override fun getName() = "Convert to assert_ne!"
+
+                            override fun getFamilyName(): String = name
+
+                            override fun applyFix(project: Project, descriptor: ProblemDescriptor) {
+                                val macro = descriptor.psiElement as RsMacroCall
+                                val assertArg = macro.assertMacroArgument!!
+                                val argPair = comparedAssertArgs(assertArg) ?: return
+
+                                val factory = RsPsiFactory(project)
+                                val newAssert = factory.createExpression("assert_ne!()") as RsCallExpr
+                                val a = newAssert.valueArgumentList.addAfter(argPair.first, newAssert.valueArgumentList.lparen)
+                                if (argPair.second != null) {
+                                    val comma = newAssert.valueArgumentList.addAfter(factory.createComma(), a)
+                                    newAssert.valueArgumentList.addAfter(argPair.second!!, comma)
+                                }
+                                macro.replace(newAssert)
+                            }
+                        }
+                    )
+                }
+                else -> return
+            }
+        }
+    }
+
+    private fun comparedAssertArgs(arg: RsAssertMacroArgument): Pair<RsExpr, RsExpr?>? {
```

Comment:
The second `RsExpr?` looks a bit off. I wonder if it would be better to enable inspection only if both of the expressions are present, and then assume that it is not null?